qdrant_client/qdrant_client.py

```diff
@@ -72,6 +72,18 @@ def __init__(
         path: Optional[str] = None,
         **kwargs: Any,
     ):
+        # Check if fastvector is installed
+        try:
+            from fastembed.qdrant_mixin import QdrantClientMixin
+
+            # If it is, add the mixin methods to this instance
+            for name, method in QdrantClientMixin.__dict__.items():
```

Comment:
dynamic assignment of methods might be problematic for IDE to provide completions and docs.
Might it be possible to define methods within QdrantClient, while keeping the implementations in the separate package?

Reply:
Those placeholder methods could have an additional instructions for the user, if the `fastembed` is not installed.

Reply:
We'd proposed the dynamic assignment because it keeps the API design/definition in `fastembed`, instead of splitting it across the client and fastembed. I agree that this means certain IDEs can trip and not register the functions properly.

We can definitely implement the functions within the client. We had agreed upon to avoid making the client bulkier in our initial conversations. Doing that would also mean, we no longer need the `fastembed` dependency at all

What do you recommend?

Reply:
Do you recommend adding 1-4 placeholder functions to the Qdrant client with the same definitions which in turn call the `fastembed` package and check for the import error?

Reply:
> Do you recommend adding 1-4 placeholder functions to the Qdrant client with the same definitions which in turn call the fastembed package and check for the import error?

Yeah, I think i would be enough. The separate package is still useful, cause it isolates heavy dependencies.

As I understood, the idea it to only have minimal additional interface for update and search. All other APIs do not require embedding inference.

Additional benefit of having it as a separate package might be interoperability. So that we define mixin base class in Qdrant Client and implement it in separate packages. In this case some other project might implement the same interface and introduce other embeddings support without pushing into our repo